src/NUnitFramework/framework/Internal/IndirectTestAssemblyAttribute.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+
+namespace NUnit.Framework.Internal
+{
+    /// <summary>
+    /// Causes the assembly to be skipped when autodiscovering tests.
+    /// Useful for tests-on-tests scenarios so that tools that run all tests in the solution do not execute the indirect tests directly.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Assembly)]
```

Comment:
@CharliePoole Because it's a framework API like any other which can be solved in one place instead of having to solve it in every tool.